File: packages/shared/src/types/address.ts
/**
 * 地址相关类型 — 镜像后端 UserAddress
 */

export interface UserAddress {
  id: string;
  userId: string;
  label: string | null;
  recipient: string;
  phone: string;
  province: string;
  city: string;
  district: string;
  address: string;
  postalCode: string | null;
  isDefault: boolean;
  createdAt: string;
  updatedAt: string;
}

File: apps/ios/Shop/Feature/User/Data/Model/Address.swift
import Foundation

struct Address: Codable, Equatable, Identifiable, Sendable {
    let id: String
    let userId: String
    let label: String?
    let recipient: String
    let phone: String
    let province: String
    let city: String
    let district: String
    let address: String
    let postalCode: String?
    let isDefault: Bool
    let createdAt: String
    let updatedAt: String

    var fullAddress: String {
        "\(province) \(city) \(district) \(address)"
    }
}

struct OrderAddress: Codable, Equatable, Sendable {
    let recipient: String
    let phone: String
    let province: String
    let city: String
    let district: String
    let address: String
    let postalCode: String?

    var fullAddress: String {
        "\(province) \(city) \(district) \(address)"
    }
}
